test/addons/openssl-client-cert-engine/test.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+const common = require('../../common');
+
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const assert = require('assert');
+const https = require('https');
+const fs = require('fs');
+const path = require('path');
+
+const engine = path.join(__dirname,
+                         `/build/${common.buildType}/testengine.engine`);
+
+const agentKey = fs.readFileSync(path.join(common.fixturesDir,
+                                           '/keys/agent1-key.pem'));
+const agentCert = fs.readFileSync(path.join(common.fixturesDir,
+                                            '/keys/agent1-cert.pem'));
+const agentCa = fs.readFileSync(path.join(common.fixturesDir,
+                                          '/keys/ca1-cert.pem'));
+
+const port = common.PORT;
+
+const serverOptions = {
+  key: agentKey,
+  cert: agentCert,
+  ca: agentCa,
+  requestCert: true,
+  rejectUnauthorized: true
+};
+
+const server = https.createServer(serverOptions, (req, res) => {
+  res.writeHead(200);
+  res.end('hello world');
+}).listen(port, common.localhostIPv4, () => {
+  const clientOptions = {
+    method: 'GET',
+    host: common.localhostIPv4,
+    port: port,
+    path: '/test',
+    clientCertEngine: engine,  // engine will provide key+cert
+    rejectUnauthorized: false, // prevent failing on self-signed certificates
+    headers: {}
+  };
+
+  const req = https.request(clientOptions, common.mustCall(function(response) {
+    let body = '';
+    response.setEncoding('utf8');
+    response.on('data', common.mustCall(function(chunk) {
```

Comment:
are we certain this will call exactly once?

Reply:
It's unnecessary to use `common.mustCall()` here in any event as the test will definitely fail if  the callback never executes, so I've removed it.